Content.Shared/Singularity/SingularitySystem.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Linq;
+using Content.Shared.Radiation;
+using Content.Shared.Singularity.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Maths;
+using Robust.Shared.Physics.Collision.Shapes;
+
+namespace Content.Shared.Singularity
+{
+    public class SingularitySystem : EntitySystem
+    {
+        private float GetFalloff(int level)
+        {
+            return level switch
+            {
+                0 => 9999f,
+                1 => 6.4f,
+                2 => 7.0f,
+                3 => 8.0f,
+                4 => 10.0f,
+                5 => 12.0f,
+                6 => 12.0f,
+                _ => -1.0f
+            };
+        }
+
+        private float GetIntensity(int level)
+        {
+            return level switch
+            {
+                0 => 0.0f,
+                1 => 2.7f,
+                2 => 14.4f,
+                3 => 47.2f,
+                4 => 180.0f,
+                5 => 600.0f,
+                6 => 800.0f,
+                _ => -1.0f
+            };
+        }
+
+        public void ChangeSingularityLevel(SharedSingularityComponent singularity, int value)
+        {
+            if (value == singularity.Level)
+            {
+                return;
+            }
+
+            value = Math.Clamp(value, 0, 6);
+
+            var physics = singularity.Owner.GetComponentOrNull<PhysicsComponent>();
+
+            if (singularity.Level > 1 && value <= 1)
+            {
+                // Prevents it getting stuck (see SingularityController.MoveSingulo)
+                if (physics != null)
+                {
+                    physics.LinearVelocity = Vector2.Zero;
+                }
+            }
+
+            singularity.Level = value;
+
+            if (singularity.Owner.TryGetComponent(out SharedRadiationPulseComponent? pulse))
+            {
+                pulse.RadsPerSecond = 10 * value;
+            }
+
+            if (singularity.Owner.TryGetComponent(out SharedAppearanceComponent? appearance))
+            {
+                appearance.SetData(SingularityVisuals.Level, value);
+            }
+
+            if (physics != null &&
+                physics.Fixtures.Any() &&
+                physics.Fixtures[0].Shape is PhysShapeCircle circle)
+            {
+                circle.Radius = value - 0.5f;
+            }
```

Comment:
This should get the fixture by name. Ideally we'd have 2 on the singularity, one for the outer pull radius and one for the inner delete radius.

Reply:
Done in https://github.com/space-wizards/space-station-14/pull/4194/commits/17db10fdaf0e540e0d58658f0565a51f5c89b3c4 except for the 2 fixture part